src/Stratis.Bitcoin.Features.Consensus/CoinviewPrefetcher.cs

```diff
@@ -42,17 +47,19 @@ public CoinviewPrefetcher(ICoinView coinview, ChainIndexer chainIndexer, ILogger
             this.chainIndexer = chainIndexer;
             this.asyncProvider = asyncProvider;
 
-            this.headersQueue = asyncProvider.CreateAndRunAsyncDelegateDequeuer<ChainedHeader>($"{nameof(CoinviewPrefetcher)}-{nameof(this.headersQueue)}", this.OnHeaderEnqueuedAsync);
+            this.headersQueue = asyncProvider.CreateAndRunAsyncDelegateDequeuer<ChainedHeader>($"{nameof(CoinviewPrefetcher)}-{nameof(this.headersQueue)}", this.OnHeaderEnqueued);
             this.coinviewHelper = new CoinviewHelper();
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
         }
 
         /// <summary>
-        /// Pre-fetches UTXOs for a block and some blocks with higher height.
-        /// pre-fetching is done on the background.
+        /// Pre-fetch UTXOs for a block and some blocks with higher height.
+        /// <para>
+        /// This task runs in the background.
+        /// </para>
         /// </summary>
-        /// <param name="header">Header of a block that is about to be partially validated and requires pre-fetching.</param>
-        private async Task OnHeaderEnqueuedAsync(ChainedHeader header, CancellationToken cancellation)
+        /// <param name="header">Header of a block that is about to be partially validated and that requires pre-fetching.</param>
+        private Task OnHeaderEnqueued(ChainedHeader header, CancellationToken cancellation)
```

Comment:
in this kind of scenarios we may want to use ValueTask instead
see https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/